src/app/components/page-header.tsx

```diff
@@ -0,0 +1,85 @@
+import { memo } from 'react';
+import { FiArrowLeft, FiX } from 'react-icons/fi';
+import { useNavigate } from 'react-router-dom';
+
+import { Box, Flex, FlexProps, IconButton } from '@stacks/ui';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { NetworkModeBadge } from '@app/components/network-mode-badge';
+import { Title } from '@app/components/typography';
+
+interface PageHeaderProps extends FlexProps {
+  actionButton?: JSX.Element;
+  closeIcon?: boolean;
+  hideActions?: boolean;
+  onClose?(): void;
+  enableClose?: boolean;
+  enableGoBack?: boolean;
+  title?: string;
+}
+export const PageHeader: React.FC<PageHeaderProps> = memo(props => {
```

Comment:
Note on name and style: 
- favour regular `function` components rather, 
- memo probably isn't necessary, it's a lightweight component
- `FC` types the component incorrectly (says there will be children when there shouldn't)
- I would also name something more specific: modal header? page header seems quite vague

Reply:
I wonder if there's a lint rule to enforce a consistent style of component, there are some cases of using React.FC it'd be nice to remove to avoid confusion

Reply:
seems it is
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/function-component-definition.md